xarray/backends/h5netcdf_.py

```diff
@@ -423,5 +424,41 @@ def open_dataset(  # type: ignore[override]  # allow LSP violation, not supporti
         )
         return ds
 
+    # TODO [MHS, 01/23/2024] This is duplicative of the netcdf4 code in an ugly way.
```

Comment:
since open_datatree shared code for both h5netcdf and netcdf4, the separation added a bit of duplication. I can/could extract this into a fnx that calls this chunk with the correct type of ncDataset, but I wasn't sure that was the right thing to do, and wasn't sure where that function should live.

Reply:
I guess it would presumably live in `xarray.backends.common.py`/`utils.py`. @kmuehlbauer any thoughts?

Reply:
I put them in common 
e4f0374